### ISSUE TITLE
`AttributeError: 'list' object has no attribute 'get'`

### ISSUE THREAD

**Issue by codepuncher** [ansible/ansible-lint#2228]
<!--- Verify first that your issue is not already reported on GitHub -->
<!--- Also test if the latest release and main branch are affected too -->

##### Summary

<!--- Explain the problem briefly below -->

Any command I run with `ansible-lint` fails with the following error

```shell
DEBUG    Logging initialized to level 10
DEBUG    Options: Namespace(cache_dir='/Users/lee-work/.cache/ansible-compat/718bee', colored=True, configured=True, cwd=PosixPath('/Users/lee-work/Code/misc/roots-trellis'), display_relative_path=True, exclude_paths=['.cache', '.git', '.hg', '.svn', '.tox'], format='rich', lintables=[], listrules=False, listtags=False, write_list=None, parseable=False, quiet=0, rulesdirs=['/usr/local/lib/python3.9/site-packages/ansiblelint/rules'], skip_list=[], tags=[], verbosity=3, warn_list=['experimental', 'role-name'], kinds=[{'jinja2': '**/*.j2'}, {'jinja2': '**/*.j2.*'}, {'yaml': '.github/**/*.{yaml,yml}'}, {'text': '**/templates/**/*.*'}, {'execution-environment': '**/execution-environment.yml'}, {'ansible-lint-config': '**/.ansible-lint'}, {'ansible-lint-config': '**/.config/ansible-lint.yml'}, {'ansible-navigator-config': '**/ansible-navigator.{yaml,yml}'}, {'inventory': '**/inventory/**.yml'}, {'requirements': '**/meta/requirements.yml'}, {'galaxy': '**/galaxy.yml'}, {'reno': '**/releasenotes/*/*.{yaml,yml}'}, {'playbook': '**/playbooks/*.{yml,yaml}'}, {'playbook': '**/*playbook*.{yml,yaml}'}, {'role': '**/roles/*/'}, {'tasks': '**/tasks/**/*.{yaml,yml}'}, {'handlers': '**/handlers/*.{yaml,yml}'}, {'vars': '**/{host_vars,group_vars,vars,defaults}/**/*.{yaml,yml}'}, {'test-meta': '**/tests/integration/targets/*/meta/main.{yaml,yml}'}, {'meta': '**/meta/main.{yaml,yml}'}, {'meta-runtime': '**/meta/runtime.{yaml,yml}'}, {'arg_specs': '**/roles/**/meta/argument_specs.{yaml,yml}'}, {'yaml': '.config/molecule/config.{yaml,yml}'}, {'requirements': '**/molecule/*/{collections,requirements}.{yaml,yml}'}, {'yaml': '**/molecule/*/{base,molecule}.{yaml,yml}'}, {'requirements': '**/requirements.yml'}, {'playbook': '**/molecule/*/*.{yaml,yml}'}, {'yaml': '**/{.ansible-lint,.yamllint}'}, {'yaml': '**/*.{yaml,yml}'}, {'yaml': '**/.*.{yaml,yml}'}], mock_modules=[], mock_roles=[], loop_var_prefix=None, var_naming_pattern=None, offline=None, project_dir='.', extra_vars=None, enable_list=[], skip_action_validation=True, rules={}, progressive=False, rulesdir=[], use_default_rules=False, config_file=None, version=False)
DEBUG    /Users/lee-work/Code/misc/roots-trellis
INFO     Set ANSIBLE_LIBRARY=/Users/lee-work/.cache/ansible-compat/718bee/modules:/Users/lee-work/.ansible/plugins/modules:/usr/share/ansible/plugins/modules
INFO     Set ANSIBLE_COLLECTIONS_PATH=/Users/lee-work/.cache/ansible-compat/718bee/collections:/Users/lee-work/.ansible/collections:/usr/share/ansible/collections
INFO     Set ANSIBLE_ROLES_PATH=/Users/lee-work/.cache/ansible-compat/718bee/roles:roles:/Users/lee-work/Code/misc/roots-trellis/vendor/roles
Traceback (most recent call last):
  File "/usr/local/bin/ansible-lint", line 8, in <module>
    sys.exit(_run_cli_entrypoint())
  File "/usr/local/lib/python3.9/site-packages/ansiblelint/__main__.py", line 252, in _run_cli_entrypoint
    sys.exit(main(sys.argv))
  File "/usr/local/lib/python3.9/site-packages/ansiblelint/__main__.py", line 165, in main
    app = get_app(offline=options.offline)
  File "/usr/local/lib/python3.9/site-packages/ansiblelint/app.py", line 252, in get_app
    _perform_mockings()
  File "/usr/local/lib/python3.9/site-packages/ansiblelint/_mockings.py", line 77, in _perform_mockings
    namespace = yaml.get("namespace", None)
AttributeError: 'list' object has no attribute 'get'

```

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

<!--- Paste verbatim output between triple backticks -->

```console (paste below)
ansible --version
ansible [core 2.13.1]
  config file = /Users/lee-work/Code/misc/roots-trellis/ansible.cfg
  configured module search path = ['/Users/lee-work/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.9/site-packages/ansible
  ansible collection location = /Users/lee-work/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.9.13 (main, May 24 2022, 21:28:31) [Clang 13.1.6 (clang-1316.0.21.2)]
  jinja version = 3.0.1
  libyaml = True



ansible-lint --version                                                                                                                                                                                                                                                                                        15:44:10
WARNING: PATH altered to include /usr/local/opt/python@3.9/bin
ansible-lint 6.3.1.dev8 using ansible 2.13.1
```

- ansible installation method: homebrew (macOS)
- ansible-lint installation method: pip3 using latest source from github

##### OS / ENVIRONMENT

<!--- Provide all relevant information below, e.g. target OS versions, network device firmware, etc. -->
macOS 12.4

##### STEPS TO REPRODUCE

<!--- Describe exactly how to reproduce the problem, using a minimal test-case -->

<!--- Paste example playbooks or commands between triple backticks below -->

```console (paste below)
~/Code/misc
[I] ➜ gh repo clone roots/trellis roots-trellis                                                                                                                                                                                                                                                                     15:46:31
Cloning into 'roots-trellis'...
remote: Enumerating objects: 9272, done.
remote: Counting objects: 100% (109/109), done.
remote: Compressing objects: 100% (53/53), done.
remote: Total 9272 (delta 60), reused 93 (delta 54), pack-reused 9163
Receiving objects: 100% (9272/9272), 1.88 MiB | 1.02 MiB/s, done.
Resolving deltas: 100% (4573/4573), done.

~/Code/misc took 4s
[I] ➜ cd roots-trellis                                                                                                                                                                                                                                                                                              15:46:57

roots-trellis on  master
[I] ➜ ansible-lint -vvv                                                                                                                                                                                                                                                                                             15:47:01
WARNING: PATH altered to include /usr/local/opt/python@3.9/bin
DEBUG    Logging initialized to level 10
DEBUG    Options: Namespace(cache_dir='/Users/lee-work/.cache/ansible-compat/718bee', colored=True, configured=True, cwd=PosixPath('/Users/lee-work/Code/misc/roots-trellis'), display_relative_path=True, exclude_paths=['.cache', '.git', '.hg', '.svn', '.tox'], format='rich', lintables=[], listrules=False, listtags=False, write_list=None, parseable=False, quiet=0, rulesdirs=['/usr/local/lib/python3.9/site-packages/ansiblelint/rules'], skip_list=[], tags=[], verbosity=3, warn_list=['experimental', 'role-name'], kinds=[{'jinja2': '**/*.j2'}, {'jinja2': '**/*.j2.*'}, {'yaml': '.github/**/*.{yaml,yml}'}, {'text': '**/templates/**/*.*'}, {'execution-environment': '**/execution-environment.yml'}, {'ansible-lint-config': '**/.ansible-lint'}, {'ansible-lint-config': '**/.config/ansible-lint.yml'}, {'ansible-navigator-config': '**/ansible-navigator.{yaml,yml}'}, {'inventory': '**/inventory/**.yml'}, {'requirements': '**/meta/requirements.yml'}, {'galaxy': '**/galaxy.yml'}, {'reno': '**/releasenotes/*/*.{yaml,yml}'}, {'playbook': '**/playbooks/*.{yml,yaml}'}, {'playbook': '**/*playbook*.{yml,yaml}'}, {'role': '**/roles/*/'}, {'tasks': '**/tasks/**/*.{yaml,yml}'}, {'handlers': '**/handlers/*.{yaml,yml}'}, {'vars': '**/{host_vars,group_vars,vars,defaults}/**/*.{yaml,yml}'}, {'test-meta': '**/tests/integration/targets/*/meta/main.{yaml,yml}'}, {'meta': '**/meta/main.{yaml,yml}'}, {'meta-runtime': '**/meta/runtime.{yaml,yml}'}, {'arg_specs': '**/roles/**/meta/argument_specs.{yaml,yml}'}, {'yaml': '.config/molecule/config.{yaml,yml}'}, {'requirements': '**/molecule/*/{collections,requirements}.{yaml,yml}'}, {'yaml': '**/molecule/*/{base,molecule}.{yaml,yml}'}, {'requirements': '**/requirements.yml'}, {'playbook': '**/molecule/*/*.{yaml,yml}'}, {'yaml': '**/{.ansible-lint,.yamllint}'}, {'yaml': '**/*.{yaml,yml}'}, {'yaml': '**/.*.{yaml,yml}'}], mock_modules=[], mock_roles=[], loop_var_prefix=None, var_naming_pattern=None, offline=None, project_dir='.', extra_vars=None, enable_list=[], skip_action_validation=True, rules={}, progressive=False, rulesdir=[], use_default_rules=False, config_file=None, version=False)
DEBUG    /Users/lee-work/Code/misc/roots-trellis
INFO     Set ANSIBLE_LIBRARY=/Users/lee-work/.cache/ansible-compat/718bee/modules:/Users/lee-work/.ansible/plugins/modules:/usr/share/ansible/plugins/modules
INFO     Set ANSIBLE_COLLECTIONS_PATH=/Users/lee-work/.cache/ansible-compat/718bee/collections:/Users/lee-work/.ansible/collections:/usr/share/ansible/collections
INFO     Set ANSIBLE_ROLES_PATH=/Users/lee-work/.cache/ansible-compat/718bee/roles:roles:/Users/lee-work/Code/misc/roots-trellis/vendor/roles
Traceback (most recent call last):
  File "/usr/local/bin/ansible-lint", line 8, in <module>
    sys.exit(_run_cli_entrypoint())
  File "/usr/local/lib/python3.9/site-packages/ansiblelint/__main__.py", line 252, in _run_cli_entrypoint
    sys.exit(main(sys.argv))
  File "/usr/local/lib/python3.9/site-packages/ansiblelint/__main__.py", line 165, in main
    app = get_app(offline=options.offline)
  File "/usr/local/lib/python3.9/site-packages/ansiblelint/app.py", line 252, in get_app
    _perform_mockings()
  File "/usr/local/lib/python3.9/site-packages/ansiblelint/_mockings.py", line 77, in _perform_mockings
    namespace = yaml.get("namespace", None)
AttributeError: 'list' object has no attribute 'get'
```

<!--- HINT: You can paste gist.github.com links for larger files -->

##### Desired Behavior

<!--- Describe what you expected to happen when running the steps above -->

To lint without this error

##### Actual Behavior

Already explained

**Comment by ssbarnea** [comment]
That tells me that you have a YAML list(sequence) instead of a dictionary(mapping0 inside your `galaxy.yml` file which is not supported.

**Comment by evgeni** [comment]
> That tells me that you have a YAML list(sequence) instead of a dictionary(mapping0 inside your `galaxy.yml` file which is not supported.

https://github.com/roots/trellis/blob/master/galaxy.yml seems to agree with you

**Comment by ssbarnea** [comment]
While the tool behaviour could be improved, the statement junk-in, junk-out also applied here. That looks like a v1 requirements.yml file renamed to galaxy.yml, so...

PS. I will keep this bug open to remember to improve the error message and avoid an exception.

**Comment by codepuncher** [comment]
Thanks for the input all. I'll deal with this in the repo with the issue.
